pkg/apis/resource/types.go

```diff
@@ -779,6 +779,12 @@ type DeviceRequestAllocationResult struct {
 	//
 	// +required
 	Device string
+
+	// AdminAccess is a copy of the AdminAccess value in the
+	// request which caused this device to be allocated.
+	//
+	// +optional
+	AdminAccess bool
```

Comment:
@klueska @byako: if you intend to support 1.31 and 1.32, then there is a slight problem: your driver cannot distinguish between "false" (no admin access with 1.32) and "not set" (behavior in 1.31, where only the spec had the original field).

One way to ensure interoperability would be to turn this into a pointer and explicitly document that "if this is not set, the corresponding field in the request in the spec needs to be checked".

This might be cleaner.

Reply:
I think it is, so I changed it: https://github.com/kubernetes/kubernetes/compare/88f8eb35dace02314007ee33e4cd73130f75e5ab..829071124985045dbeee84b4b03feac89b024968

Reply:
Why would we need to distinguish? As long as we know when it is True - that's what matters, isn't it?

Reply:
Making it a pointer enables clients to detect "old" allocations where admin access was defined only in the spec.

If a client doesn't care to support 1.31, it can always use `ptr.Deref(, false)`.

Reply:
Upgrades are allowed to break when alpha features are involved, but that doesn't mean that they are automatically unsupported or discouraged. If it's possible to support them, then it's better to do so.

The main practical purpose is to enable @klueska to support admin access in a driver release that then works with 1.31 and 1.32.

Doing it "cleanly" according to API guidelines is less important in practice. We could pretend that 1.31 doesn't exist and just use `AdminAccess bool` like I did initially. It certainly was easier (no need for validation, no code changes in various places to set the field in new allocations).

Reply:
I changed the wording again a bit by adding "... if the field is unset.".

@kannon92, @byako, @klueska: is this good now for API review? Someone please LGTM.